src/Network/GRPC/MQTT/Message/Packet.hs

```diff
@@ -221,21 +257,26 @@ makePacketSender limit publish message = do
 
   if fromIntegral (ByteString.length message) <= maxPayloadSize
     then do
-      -- Perform a single publish on the current thread in case that the 
+      -- Perform a single publish on the current thread in case that the
       -- @message@ length is less than the maximum payload size.
       let packet = Packet message 0 1
       publish (wireWrapPacket' packet)
     else do
       caps <- liftIO getNumCapabilities
       jobs <- newTVarIO 0
 
-      replicateConcurrently_ caps $ fix \next ->
-        atomically (takeJobId jobs) >>= \case
-          Nothing -> pure ()
-          Just jobid -> do
-            let packet = makePacket jobid
-            publish (wireWrapPacket' packet)
-            next
+      replicateConcurrently_ caps do
+        let bufferSize :: Int
+            bufferSize = fromIntegral (limit + minPacketSize)
```

Comment:
It is not clear to me why `minPacketSize` is both added to `limit` to get the output buffer size and also subtracted from `limit` to get the maximum `Packet` `payload` size, and never used as-is.  One should only need to do one or the other.